src/App.tsx

```diff
@@ -1,42 +1,36 @@
-import React from 'react';
+import React, { useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 import './App.scss';
+import { Tab } from './types/Tab';
+import { Tabs } from './components/Tabs';
 
 export const tabs = [
   { id: 'tab-1', title: 'Tab 1', content: 'Some text 1' },
   { id: 'tab-2', title: 'Tab 2', content: 'Some text 2' },
   { id: 'tab-3', title: 'Tab 3', content: 'Some text 3' },
 ];
 
+const firstTab: Tab | undefined = tabs[0];
+
 export const App: React.FC = () => {
+  const [selectedTab, setSelectedTab] = useState(firstTab);
+
+  const selectTab = (tab: Tab) => {
+    setSelectedTab(tab);
+  };
```

Comment:
Let's use `useCallback` for rendering optimisation, cause you are passing the callback to the another component